modulus

```diff
@@ -196,6 +204,25 @@ truncate_bin() {
     local END=$(gdisk -l "$IN" | grep ROOT | awk '{print $3}')
     dd if="$IN" of="$IN" bs=512 skip="$START" conv=notrunc
     truncate -s $(((END-4096+1)*512)) "$IN"
+    resize_image_vol
+}
+
+resize_image_vol() {
+    # shellcheck disable=SC2155
+    local BYTES_AVAILABLE=$(df --output=avail -B 1 . | tail -1)
+    # shellcheck disable=SC2155
+    local TOTAL_BYTES_FOR_IMAGE=$((IMAGE_VOLSIZE_GIB*1073741824))
+    if ! [ "$TOTAL_BYTES_FOR_IMAGE" -lt "$BYTES_AVAILABLE" ]; then
+        echo "err: not enough disk space. $TOTAL_BYTES_FOR_IMAGE-byte image volume size requested but there are only $BYTES_AVAILABLE bytes availble on-disk"
+        exit 1
+    fi
+    truncate -s "$TOTAL_BYTES_FOR_IMAGE" "$IN"
```

Comment:
The variable `$IN` is local to the function where it is defined. It should not be in this functions’s scope

Reply:
`resize_image_vol()` is a child function of `truncate_bin()` which is why `$IN` as a local variable works, but sure, I can be more explicit about passing that arg in

Reply:
So, would you like me to redeclare it as a local variable named `$IN` within `resize_image_vol()` as well, or is that too redundant? Or just leave as is?

Reply:
Yes please pass the file path as an argument to the func so that we can reuse the function and not depend on the scope of the parent function.